packages/api-server/api_server/routes/fleets.py

```diff
@@ -68,21 +68,23 @@ async def get_robots(
                 # multiple fleets with the same robot name and there are active tasks
                 # assigned to those robots and the robot states are not synced to the
                 # tasks summaries.
-                if r is None:
+                if r is None and t.fleet_name == "":
```

Comment:
Can you explain how this fixes the bug? As far as I see, if `fleet_name` is "", `r` should always be `None` as a robot must always belong to a fleet.